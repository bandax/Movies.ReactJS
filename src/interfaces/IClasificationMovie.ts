export interface IClasification {
    id: string;
    name: string;
}
