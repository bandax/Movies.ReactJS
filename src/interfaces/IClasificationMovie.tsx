export interface IClasification {
    id: string,
    name: string
}

export interface IClasificationMovieProps {
    clasificationTypes: IClasification[],    
}

