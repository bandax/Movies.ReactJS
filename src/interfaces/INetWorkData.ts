export interface INetworkData {
  url: string;
  body: string;
  contentType: string;
}
