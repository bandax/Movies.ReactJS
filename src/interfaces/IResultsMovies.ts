export interface IMovie {
    id: string;
    title: string;
    image: string;
    year: number;
    categories: ICategory[];
    releaseDate: string;
    review: string;
    url: string;    
    runtime: string;
}

export interface ICategory {
    id: string; 
    name: string;
}
