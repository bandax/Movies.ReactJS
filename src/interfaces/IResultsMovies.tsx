export interface IMovie {
    id: string,
    title: string,
    image: string,
    year: number,
    categories: ICategory[]
}

export interface ICategory {
    id: string, 
    name: string
}

export interface IResultsMovie {
    total: number,   
    movies: IMovie[]
}

export interface IResultsMovieProps {
    title: string,
    resultsMovies: IResultsMovie,    
}


export interface IDetailsMovieProps {
    movie: IMovie   
}
