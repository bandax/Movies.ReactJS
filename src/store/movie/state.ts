import { IMovie } from "../../interfaces/IResultsMovies";

export interface MovieState {
  movies: IMovie[];
  loading: boolean;
  errorMessage: string;
}
