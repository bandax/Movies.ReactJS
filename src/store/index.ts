import { combineReducers, applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { movieReducer } from './movie/reducers';

const rootReducer = combineReducers({
  movie: movieReducer,
});

export type RootState = ReturnType<typeof rootReducer>;

export const store = createStore(rootReducer, applyMiddleware(thunk));
