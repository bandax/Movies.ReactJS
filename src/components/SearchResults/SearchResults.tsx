import * as React from 'react';
import { useLocation } from 'react-router-dom';
import { IMovieData } from '../../interfaces/IMovieData';
import { DetailsMovie } from '../DetailsMovie/DetailsMovie';
import { connect } from 'react-redux';
import { RootState } from '../../store/index';
import { searchMovies } from './thunks';
import { ThunkDispatch } from 'redux-thunk';
import { getMoviesLoading, getMoviesData } from '../ResultsMovie/selectors';
import { Action } from 'redux';

const mapStateToProps = (state: RootState) => ({
  movie: state.movie.movie,
  movies: getMoviesData(state),
  isLoading: getMoviesLoading(state),
});

const mapDispatchToProps = (
  dispatch: ThunkDispatch<RootState, void, Action>
) => {
  return {
    searchMovies: (term: string) => dispatch(searchMovies(term)),
  };
};

type Props = ReturnType<typeof mapStateToProps> &
  ReturnType<typeof mapDispatchToProps>;

const SearchResults: React.FunctionComponent<Props> = ({
  movie,
  movies,
  searchMovies,
  isLoading,
}) => {
  const params = new URLSearchParams(useLocation().search);
  const term = params.get('term');
  React.useEffect(() => {
    console.log(`searching ${term}`);
    searchMovies(term);
  }, []);

  const handleEditMovie = function (movie: IMovieData) {
    console.log('tests');
  };

  const handleDeleteMovie = function (movie: IMovieData) {
    console.log('tests');
  };

  const loadingMessage = <div>Searching movies...</div>;
  const content = (
    <>
      <div className="app">Results for {term}</div>
      <div className="list-movies">
        <div className="total-result">
          <span className="found-label">
            <b>{movies.length}</b> movies found
          </span>
        </div>
        <div className="display-movie">
          {movies.map((movie: IMovieData) => (
            <DetailsMovie
              key={movie.id}
              movie={movie}
              onEditMovie={handleEditMovie}
              onDeleteMovie={handleDeleteMovie}
            />
          ))}
        </div>
      </div>
    </>
  );

  return isLoading ? loadingMessage : content;
};

export default connect(mapStateToProps, mapDispatchToProps)(SearchResults);
