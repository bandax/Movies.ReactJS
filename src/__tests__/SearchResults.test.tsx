import * as React from 'react';
import SearchResults from '../components/SearchResults/SearchResults';
import thunk, { ThunkDispatch } from 'redux-thunk';
import {
  render,
  fireEvent,
  waitFor,
  screen,
  getByText,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom/extend-expect';
import clasificationTypes from '../data/clasifications.json';
import { IMovieData } from '../interfaces/IMovieData';
import {
  HashRouter as Router,
  Route,
  Switch,
  MemoryRouter,
} from 'react-router-dom';

import configureMockStore, { MockStoreCreator } from 'redux-mock-store';
import { MovieState } from '../store/movie/state';
import { Action } from 'redux';
import { Provider } from 'react-redux';
import { RootState } from '../store/index';

const movies: IMovieData[] = [
  {
    id: 1234,
    title: 'Avengers Infinity War',
    poster_path:
      'https://upload.wikimedia.org/wikipedia/commons/thumb/d/d7/Scarymovie4-logo.svg/1920px-Scarymovie4-logo.svg.png',
    release_date: new Date().toLocaleDateString(),
    budget: 123,
    revenue: 123,
    vote_average: 123,
    vote_count: 123,
    runtime: 412,
    tagline: '12343',
    overview: 'tests',
    genres: ['Action & Adventure', 'Comedy'],
  },
  {
    id: 1235,
    title: 'Avengers Infinity War II',
    poster_path:
      'https://upload.wikimedia.org/wikipedia/commons/thumb/d/d7/Scarymovie4-logo.svg/1920px-Scarymovie4-logo.svg.png',
    release_date: new Date().toLocaleDateString(),
    budget: 123,
    revenue: 123,
    vote_average: 123,
    vote_count: 123,
    runtime: 412,
    tagline: '12343',
    overview: 'tests',
    genres: ['Action & Adventure', 'Comedy'],
  },
];

type DispatchExts = ThunkDispatch<MovieState, void, Action>;
const middleware = [thunk];
const state: RootState = {
  movie: {
    movie: null,
    errorMessage: '',
    findMovies: true,
    loading: false,
    movies: movies,
  },
};

const mockStore = configureMockStore<RootState, DispatchExts>(middleware);

describe('<SearchResults />', () => {
  let store;

  beforeEach(() => {
    store = mockStore(state);
    jest.mock('react-router-dom', () => ({
      ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
      useParams: () => ({
        searchQuery: 'Infinity',
      }),
    }));
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render a SearchResults Component', async () => {
    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/search/Infinity']}>
          <Route path="/search/:searchQuery">
            <SearchResults />
          </Route>
        </MemoryRouter>
      </Provider>,
    );

    const fragment = asFragment();

    expect(fragment).toMatchSnapshot();
  });

  it('should render label with keyword to search', async () => {
    const { getByText } = render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/search/Infinity']}>
          <Route path="/search/:searchQuery">
            <SearchResults />
          </Route>
        </MemoryRouter>
      </Provider>,
    );

    const searchWord = getByText('Results for Infinity');
    screen.debug();

    expect(searchWord).toBeInTheDocument();
  });
});
