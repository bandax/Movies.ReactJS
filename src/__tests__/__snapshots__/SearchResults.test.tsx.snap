// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResults /> should render a SearchResults Component 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    Results for Infinity
  </div>
  <div
    class="list-movies"
  >
    <div
      class="total-result"
    >
      <span
        class="found-label"
      >
        <b>
          2
        </b>
         movies found
      </span>
    </div>
    <div
      class="display-movie"
    >
      <div
        class="details-movie"
      >
        <img
          class="poster-movie"
          src="https://upload.wikimedia.org/wikipedia/commons/thumb/d/d7/Scarymovie4-logo.svg/1920px-Scarymovie4-logo.svg.png"
        />
        <br />
        <span
          class="title-movie"
        >
          Avengers Infinity War
        </span>
        <span
          class="year-movie"
        >
          2020
        </span>
        <span
          class="categories-movie"
        >
          Action & Adventure, Comedy
        </span>
        <button
          class=""
        >
          Edit Movie
        </button>
        <button
          class=""
        >
          Delete Movie
        </button>
        <a
          href="#/film/1234"
        >
          Show Details Movie
        </a>
      </div>
      <div
        class="details-movie"
      >
        <img
          class="poster-movie"
          src="https://upload.wikimedia.org/wikipedia/commons/thumb/d/d7/Scarymovie4-logo.svg/1920px-Scarymovie4-logo.svg.png"
        />
        <br />
        <span
          class="title-movie"
        >
          Avengers Infinity War II
        </span>
        <span
          class="year-movie"
        >
          2020
        </span>
        <span
          class="categories-movie"
        >
          Action & Adventure, Comedy
        </span>
        <button
          class=""
        >
          Edit Movie
        </button>
        <button
          class=""
        >
          Delete Movie
        </button>
        <a
          href="#/film/1235"
        >
          Show Details Movie
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
